ui/components/ui/editable-label/editable-label.js

```diff
@@ -19,91 +20,76 @@ class EditableLabel extends Component {
-    const { accountsNames } = this.props;
+    const { isValidAccountName, errorMessage } = getAccountNameErrorMessage(
+      this.props.accounts,
+      this.context,
+      this.state.value,
+      this.props.defaultValue,
+    );
 
-    return [
-      <input
-        key={1}
-        type="text"
-        required
-        dir="auto"
-        value={this.state.value}
-        onKeyPress={(event) => {
-          if (event.key === 'Enter') {
-            this.handleSubmit();
-          }
-        }}
-        onChange={(event) => this.setState({ value: event.target.value })}
-        data-testid="editable-input"
-        className={classnames('large-input', 'editable-label__input', {
-          'editable-label__input--error':
-            value === '' || accountsNames.includes(value),
-        })}
-        autoFocus
-      />,
-      <button
-        className="editable-label__icon-button"
-        key={2}
-        onClick={() => this.handleSubmit()}
-      >
-        <i className="fa fa-check editable-label__icon" />
-      </button>,
-    ];
+    return (
+      <div className={classnames('editable-label', this.props.className)}>
+        <input
+          key={1}
+          type="text"
+          required
+          dir="auto"
+          value={this.state.value}
+          onKeyPress={(event) => {
+            if (event.key === 'Enter') {
+              this.handleSubmit(isValidAccountName);
+            }
+          }}
+          onChange={(event) => this.setState({ value: event.target.value })}
+          data-testid="editable-input"
+          className={classnames('large-input', 'editable-label__input', {
+            'editable-label__input--error': !isValidAccountName,
+          })}
+          autoFocus
+        />
+        <button
+          className="editable-label__icon-button"
+          key={2}
+          onClick={() => this.handleSubmit(isValidAccountName)}
+        >
+          <i className="fa fa-check editable-label__icon" />
+        </button>
+        <div className="editable-label__error editable-label__error-amount">
+          {errorMessage}
+        </div>
+      </div>
+    );
   }
 
   renderReadonly() {
-    return [
-      <div key={1} className="editable-label__value">
-        {this.state.value}
-      </div>,
-      <button
-        key={2}
-        className="editable-label__icon-button"
-        data-testid="editable-label-button"
-        onClick={() => this.setState({ isEditing: true })}
-      >
-        <i className="fas fa-pencil-alt editable-label__icon" />
-      </button>,
-    ];
-  }
-
-  render() {
-    const { isEditing, value } = this.state;
-    const { className, accountsNames } = this.props;
-
     return (
-      <>
-        <div className={classnames('editable-label', className)}>
-          {isEditing ? this.renderEditing() : this.renderReadonly()}
+      <div className={classnames('editable-label', this.props.className)}>
+        <div key={1} className="editable-label__value">
```

Comment:
what is the purpose of the key attributes here and below?

Reply:
Oh I see, previously these components were in an array. Looks like they can be removed now?

Reply:
Yes, you're right, the `key` can be removed.